README.md

```diff
@@ -13,3 +13,35 @@ Contents of this repo:
 ## License
 
 (c) Google, 2015. Licensed under [Apache-2](LICENSE) license.
+
+## Usage
+
+Optionally, if you would not like `goacme` to generate the account key for you, you should create an unencrypted private key for use with account registration. This can be done using OpenSSL:
+
+```
+mkdir -p ~/.config/acme
+openssl -out ~/.config/acme/account.key 2048
+```
+
+Next, register an account using the `goacme` client. The contact information, such as email address at the end, is optional but recommended. If you would like to have `goacme` generate the account key for you instead of using OpenSSL or similar software, include the `-gen` flag. Both example are shown below, though only one should be run depending on the whether you are providing the account key or having `goacme` generate one.
+
+```
+# Manually Generated Account Key
+goacme reg mailto:your-email-address@gmail.com
+
+# Automatically Generated Account Key
+goacme reg -gen mailto:your-email-address@gmail.com
+```
+
+Next, review the terms and if you agree, you can accept them like so:
+
+```
+goacme update -accept
+```
+
+Now, we are ready to request our certificate. To do this, we need a private key for the domain first.
+
+```
+openssl -out ~/.config/acme/www.domain.com.key 2048
+goacme cert www.domain.com
```

Comment:
sounds good. a few things:

1. did you mean "openssl genrsa -out ... 2048"? (missing genrsa argument);
same for account key above, in the registration phase.
2. for auto-generated, there's no need for -k ~/config/... argument.
3. I would remove the whole sentence mentioning key location. like you
said, it's a "getting started" short version. if users want to get more
details, they can always do that by installing goacme and executing "goacme
help cert".

On 9 June 2016 at 19:18, Kevin Kirsche <notifications@github.com> wrote:

> In README.md
> <https://github.com/google/goacme/pull/27#discussion_r66494030>:
>
> > +
> > +# Automatically Generated Account Key
> > +goacme reg -gen mailto:your-email-address@gmail.com
> > +```
> > +
> > +Next, review the terms and if you agree, you can accept them like so:
> > +
> > +```
> > +goacme update -accept
> > +```
> > +
> > +Now, we are ready to request our certificate. To do this, we need a private key for the domain first.
> > +
> > +```
> > +openssl -out ~/.config/acme/www.domain.com.key 2048
> > +goacme cert www.domain.com
>
> Ah, I didn't know that. I'll see if that's fixable after we get through
> this PR. Does this seem better (main paragraph not indented to make it
> easier to read):
>
> Now, we are ready to request our certificate. The certificate will be
> placed alongside key file, specified with -k argument. If the key file
> does not exist, a new one will be created. Default location for the key
> file is ~/.config/acme/domain.key, where domain is the actually domain
> name provided as the command argument, in this, it would be
> ~/.config/acme/example.com.key. If you would prefer to generate the key
> yourself, this may be done using openssl or a similar tool, and has been
> included / shown in the example below.
>
> # Manually Generated Private Domain Key
> openssl -out ~/.config/acme/example.com.key 2048
> goacme cert example.com
>
> # Automatically Generated Private Domain Key
> goacme cert -k ~/.config/acme/example.com.key example.com
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/goacme/pull/27/files/f99bc3fc5e8a3a1b822c8d7a0d866f57770c6195#r66494030>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABjPX7zqckkbsjcTIvpjGvZjT3EUKjCks5qKFj3gaJpZM4IsnSt>
> .
>